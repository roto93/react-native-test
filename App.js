import { StatusBar } from 'expo-status-bar';
import React, { useState } from 'react';
import { StyleSheet, Text, View, TextInput, TouchableOpacity } from 'react-native';
import Article from './src/components/Article';

export default function App() {
  const [color, setColor] = useState();
  return (
    <View style={styles.container}>
      <TouchableOpacity testID="btn-blue" style={styles.TO}
        onPress={() => { setColor('lightblue') }}>
        <Text>blue</Text>
      </TouchableOpacity>
      <TouchableOpacity testID="btn-tomato" style={styles.TO}
        onPress={() => { setColor('tomato') }}>
        <Text>tomato</Text>
      </TouchableOpacity>
      <Article color={color} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  TO: {
    borderWidth: 1,
    width: 120,
    height: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
  t: {
    color: 'lightblue',
  }
});
