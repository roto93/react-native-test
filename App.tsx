import { StatusBar } from 'expo-status-bar';
import React, { useState, useRef, useEffect } from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
import Svg, { Path } from 'react-native-svg'
import Animated, { useSharedValue, useAnimatedProps, withTiming, withSequence, withDelay, Easing } from 'react-native-reanimated'
export default function App() {
    const AnimatedPath = Animated.createAnimatedComponent(Path)

    const [length1, setLength1] = useState(0);
    const [length2, setLength2] = useState(0);
    const ref1 = useRef(null)
    const ref2 = useRef(null)
    const progress1 = useSharedValue(0)
    const progress2 = useSharedValue(0)

    const path1 = 'M 10 10 L 200 10 '
    const path2 = 'M 10 100 L 200 100'
    const animatedProps1 = useAnimatedProps(() => {
        return {
            strokeDashoffset: length1 - length1 * progress1.value
        }
    })
    const animatedProps2 = useAnimatedProps(() => {
        return {
            strokeDashoffset: length2 - length2 * progress2.value
        }
    })
    useEffect(() => {
        progress1.value = withTiming(1, { easing: Easing.linear })
        progress2.value = withDelay(300, withTiming(1, { easing: Easing.linear }))
    }, [])

    return (
        <View style={styles.container}>
            <Svg width={300} height={200} viewBox={'0 0 473 217'}>
                <AnimatedPath animatedProps={animatedProps1} ref={ref1} onLayout={() => { setLength1(ref1.current.getTotalLength()) }} stroke='black' d={path1} strokeWidth={4} strokeDasharray={`${length1} 1000`} strokeDashoffset={length1} />
                <AnimatedPath animatedProps={animatedProps2} ref={ref2} onLayout={() => { setLength2(ref2.current.getTotalLength()) }} stroke='black' d={path2} strokeWidth={4} strokeDasharray={`${length2} 1000`} strokeDashoffset={length2} />
            </Svg>
            <Button title='test' onPress={() => {
                progress1.value = withTiming(1)
                progress2.value = withTiming(1)
            }}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
});
