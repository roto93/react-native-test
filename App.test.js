import React from 'react'
import App from "./App";
import { render, cleanup, fireEvent } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

afterEach(cleanup);

test("snapshot", () => {
    const app = renderer.create(<App />).toJSON()
    expect(app).toMatchSnapshot()
})

test("can button1 change text color in Article?", () => {
    const { getByTestId } = render(<App />)
    const el = getByTestId('btn-blue')

    fireEvent.press(el)
    const article1 = getByTestId('article1')

    expect(article1.props.style.color).toBe('lightblue')
})

test("can button2 change text color in Article?", () => {
    const { getByTestId } = render(<App />)
    const el = getByTestId('btn-tomato')

    fireEvent.press(el)
    const article1 = getByTestId('article1')

    expect(article1.props.style.color).toBe('tomato')
})