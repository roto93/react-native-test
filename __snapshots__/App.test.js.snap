// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "opacity": 1,
        "width": 120,
      }
    }
    testID="btn-blue"
  >
    <Text>
      blue
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "opacity": 1,
        "width": 120,
      }
    }
    testID="btn-tomato"
  >
    <Text>
      tomato
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#e3e3e3",
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": undefined,
        }
      }
      testID="article1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </Text>
  </View>
</View>
`;
